src/main/java/com/spotify/docker/client/messages/swarm/Service.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.docker.client.messages.swarm;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+import java.util.Date;
+import java.util.Objects;
+
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.ANY;
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.NONE;
+
+@JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
+public class Service {
+
+    @JsonProperty("ID")
+    private String id;
+
+    @JsonProperty("Version")
+    private Version version;
+
+    @JsonProperty("CreatedAt")
+    private Date createdAt;
+
+    @JsonProperty("UpdatedAt")
+    private Date updatedAt;
+
+    @JsonProperty("Spec")
+    private ServiceSpec spec;
+
+    @JsonProperty("Endpoint")
+    private Endpoint endpoint;
+
+    @JsonProperty("UpdateStatus")
+    private UpdateStatus updateStatus;
+
+    public String id() {
+        return id;
+    }
+
+    public Version version() {
+        return version;
+    }
+
+    public Date createdAt() {
+    return createdAt == null ? null : new Date(createdAt.getTime());
+}
+
+    public Date updatedAt() {
+    return updatedAt == null ? null : new Date(updatedAt.getTime());
+}
+
+    public ServiceSpec spec() {
+        return spec;
+    }
+
+    public Endpoint endpoint() {
+        return endpoint;
+    }
+
+    public UpdateStatus updateStatus() {
+        return updateStatus;
+    }
+
+    public static class Criteria {
+
+        /** Filter by service id */
+        String serviceId;
+
+        /** Filter by service name */
+        String serviceName;
+
+        public String getServiceId() {
+            return serviceId;
+        }
+
+        public void setServiceId(String serviceId) {
+            this.serviceId = serviceId;
+        }
+
+        public String getServiceName() {
+            return serviceName;
+        }
+
+        public void setServiceName(String serviceName) {
+            this.serviceName = serviceName;
+        }
+    }
+
+    public static class CriteriaBuilder {
```

Comment:
For consistency, can you use the builder pattern in `ContainerConfig`? Also make sure the access modifiers for classes and methods are as strict as they can be.